### PR TITLE
Working prototype for the demo 

### PR DESCRIPTION
**Pull request by snakedye**
* https://github.com/ai-cfia/fertiscan-backend/issues/11
* https://github.com/ai-cfia/fertiscan-backend/issues/12
* Closes https://github.com/ai-cfia/fertiscan-backend/issues/8
* Closes https://github.com/ai-cfia/fertiscan-backend/issues/7
* Closes https://github.com/ai-cfia/fertiscan-backend/issues/9

Once this PR is merged it should serve as the reference for building the front-end.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
+# server.py
+
+from flask import Flask, request, jsonify, render_template
+from werkzeug.utils import secure_filename
+import os
+import json
+from io import BytesIO
+from dotenv import load_dotenv
+from backend import DocumentStore, OCR, LanguageModel
```
Comment (ibrahim-kabir) [inline app.py]: Import orders 👆🏻

File: app.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+# Load environment variables
+load_dotenv()
+
+# Ensure the directory for uploaded images exists
+UPLOAD_FOLDER = 'uploads'
```
Comment (ibrahim-kabir) [inline app.py:12]: Should be an environment variable

File: backend/document_store.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# fertiscan/io/document_storage.py
+
+import os
+
+class DocumentStore:
```
Comment (ibrahim-kabir) [inline backend/document_store.py:3]: Bad naming, DocumentStore refers to a market.
```suggest
class DocumentStorage:
```

File: backend/document_store.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        if isinstance(page, str):
+            if not os.path.exists(page):
+                raise FileNotFoundError(f"The file {page} does not exist.")
+            file = open(page)
+            image = file.read()
+            file.close()
+        elif isinstance(page, bytes):
+            image = page
+        else:
+            raise ValueError("The document must be a file path (str) or in bytes format.")
```
Comment (ibrahim-kabir) [inline backend/document_store.py:17]: Nice, error handling 👍 !

File: backend/document_store.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    def __init__(self):
+        self.document = []
+
+    def add_page(self, page):
+        if isinstance(page, str):
+            if not os.path.exists(page):
+                raise FileNotFoundError(f"The file {page} does not exist.")
+            file = open(page)
+            image = file.read()
+            file.close()
```
Comment (ibrahim-kabir) [inline backend/document_store.py:13]: Try/Except would be more suited in this case.
> You should do what you expect to work. If it doesn’t work and an exception happens, then just catch the exception and handle it appropriately.
[Source](https://realpython.com/python-lbyl-vs-eafp/#:~:text=The%20%E2%80%9CEasier%20to%20Ask%20Forgiveness%20Than%20Permission%E2%80%9D%20(EAFP)%20Style,-Grace%20Murray%20Hopper&text=It%20suggests%20that%20right%20away,exception%20and%20handle%20it%20appropriately.&text=In%20Python%2C%20the%20EAFP%20coding%20style%20is%20pretty%20popular%20and%20common.)

File: backend/language_model.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+# fertiscan/llm/language_model.py
+
+from openai import AzureOpenAI
+
+prompt_file = open('./prompt.txt')
```
Comment (ibrahim-kabir) [inline backend/language_model.py:3]: prompt.txt should be an environment variable. Such as "PROMPT_PATH"

File: backend/language_model.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+SETUP_PROMPT = prompt_file.read()
+prompt_file.close()
```
Comment (ibrahim-kabir) [inline backend/language_model.py:5]: Why are you doing this ? Whis isn't this into the class ?

File: requirements.txt [hunk 8]
```diff
@@ -1,2 +1,4 @@
 azure-ai-formrecognizer==3.3.0
-flask
\ No newline at end of file
+flask
+dotenv
+openai
```
Comment (ibrahim-kabir) [inline requirements.txt:4]: Add a newline

File: tests/test_document_store.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
+import unittest
+from unittest.mock import patch, mock_open, MagicMock
+from io import BytesIO
+import os
+from backend import DocumentStore  # Replace 'your_module_path' with the actual module path
```
Comment (ibrahim-kabir) [inline tests/test_document_store.py:5]: Order imports

**Review by ibrahim-kabir — CHANGES_REQUESTED** [review]
**Checks (workflows) are not working** ❌
Add checks for markdown
README.md is empty, add **at least** the setup procedure for development and explanation about this repo.

**Review by SamuelPelletierEvraire — APPROVED** [review]
All good, just make the request that ibrahim ask

**Comment by SamuelPelletierEvraire** [comment]
For the workflow, we are going to fix this after the demo because we don't have time for now

**Pull request merged**